src/python/pants/backend/go/util_rules/sdk.py

```diff
@@ -82,9 +101,14 @@ async def go_sdk_invoke_setup(goroot: GoRoot) -> GoSdkRunSetup:
             f"""\
             export GOROOT={goroot.path}
             sandbox_root="$(/bin/pwd)"
-            export GOPATH="${{sandbox_root}}/gopath"
+            export GOPATH="${{sandbox_root}}/{SANDBOX_CACHE}"
             export GOCACHE="${{sandbox_root}}/cache"
-            /bin/mkdir -p "$GOPATH" "$GOCACHE"
+            # go install targets GOBIN, updated here to keep binaries outside of the cached GOPATH
+            export GOBIN="${{sandbox_root}}/gobin"
```

Comment:
Does it make sense to only set `GOBIN` when running `go install`?

Reply:
I'd lean to have it always set - avoid the complexity of conditional logic in the bash script, and multiple places where installed binaries might end up.

Because there's no additional `mkdir` call needed (go install creates the path when called) - I wasn't seeing any extra impact on the sandboxes, just an extra environment variable that's usually ignored.  Let me know if there are other cases or impacts to consider though.

Reply:
Makes sense.